skiko/src/commonMain/kotlin/org/jetbrains/skia/paragraph/TextStyle.kt

```diff
@@ -110,10 +110,24 @@ class TextStyle internal constructor(ptr: NativePointer) : Managed(ptr, _Finaliz
         }
     }
 
-    var decorationStyle: org.jetbrains.skia.paragraph.DecorationStyle
+    var decorationStyle: DecorationStyle
         get() = try {
             Stats.onNativeCall()
-            _nGetDecorationStyle(_ptr)
+            val decorationStyleData = withResult(IntArray(4)) {
```

Comment:
Maybe it will be better to move deserialization into `DecorationStylez, like `Rect.Companion.fromInteropPointer`